app/models/beta/search/intercept_message.rb

```diff
@@ -62,30 +78,52 @@ def build(search_query)
           intercept_message = new
           intercept_message.term = normalised_query
           intercept_message.message = normalise_message(message)
+          intercept_message.generate_references_and_formatted_message!
           intercept_message
         end
 
         def intercept_messages
-          @intercept_messages ||= YAML.load_file(INTERCEPT_MESSAGES_SOURCE_PATH)
+          @intercept_messages ||= YAML.load_file(INTERCEPT_MESSAGES_SOURCE_PATH).transform_keys(&method(:normalise_query))
+        end
+
+        def all_references
+          @all_references ||= intercept_messages.keys.map(&method(:build)).each_with_object(Hash.new('')) do |intercept_message, acc|
+            acc.merge!(intercept_message.references) do |_reference_id, previous_terms, new_term|
+              merged = "#{previous_terms} #{new_term}"
+              merged.scan(/\w+/).join(' ').downcase
+            end
+          end
         end
 
         private
 
         def normalise_query(search_query)
-          search_query.downcase.scan(/\w+/).join(' ')
+          search_query.downcase.scan(/\w+/).reject { |term| TradeTariffBackend.stop_words.include?(term) }.join(' ')
```

Comment:
Extend test example to include some stop words that get removed/matched on an intercept message